### DISCUSSION TITLE
Need some regex help

### DISCUSSION THREAD

**Discussion by wolfgang-ch** [mytourbook/mytourbook#1005] [Q&A]
I'm currently implementing a feature to copy a tour by also adding an incremented counter at the newly created tour title. The counter is incremented with regex which I found [here](https://stackoverflow.com/questions/12941362/is-it-possible-to-increment-numbers-using-regex-substitution/12946132#answer-12946132)

![image](https://user-images.githubusercontent.com/1283445/222657250-d3496c04-66c6-4d42-9fc1-6e922cd9aabe.png)

This regex is working for single digit values but for the other, it is adding or inserting a `1`. With my regex skills I could not find the reason why it is not working. 

This regex is working with Javascript and can be tested in https://regex101.com/r/XT8Rmb/1 but this regex is not working with Java. 

The upper 4 lines are from Javascript the lower 4 lines are from Java

![image](https://user-images.githubusercontent.com/1283445/222658244-c365d6d3-b0c2-4428-83c3-7e4733866d31.png)

The tour copy feature can be tested with this commit https://github.com/mytourbook/mytourbook/commit/662337b447e98fc089535d9340546da17814e593

Can someone solve this problem or has Java a limitation for this regex or is it a Java bug?

This code is used very rarely and is not mission critical but it would be still interresting why it is not working, maybe for other usages.

**Comment by FJBDev** [comment]
Did you try to use the official [Java RegEx](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/regex/package-summary.html) package/methods ?

**Reply by wolfgang-ch** [reply]
The code

![image](https://user-images.githubusercontent.com/1283445/223655120-34b40965-b300-40c9-a6e1-c2a20398d8ef.png)

is using the String replaceAll method which is using

![image](https://user-images.githubusercontent.com/1283445/223655555-f23951df-658c-4ca8-95f2-bc01d286ef2c.png)
